bunq/sdk/exception.py

```diff
@@ -1,44 +1,22 @@
 class ApiException(Exception):
-    """
-    :type _response_code: int
-    """
-
-    # Constants for formatting messages
-    _FORMAT_RESPONSE_CODE_LINE = 'HTTP Response Code: {}'
-    _GLUE_ERROR_MESSAGES = '\n'
-
-    def __init__(self, response_code, messages):
+    def __init__(self, message, response_code):
         """
+        :type message: str
         :type response_code: int
-        :type messages: list[str]
         """
 
-        super(ApiException, self).__init__(
-            self.generate_message_error(response_code, messages)
-        )
         self._response_code = response_code
+        self._message = message
 
-    def generate_message_error(self, response_code, messages):
-        """
-        :type response_code: int
-        :type messages: list[str]
-
-        :rtype: str
-        """
-
-        line_response_code = self._FORMAT_RESPONSE_CODE_LINE \
-            .format(response_code)
-
-        return self.glue_messages([line_response_code] + messages)
+        super(ApiException, self).__init__(message)
 
-    def glue_messages(self, messages):
+    @property
+    def message(self):
```

Comment:
Can't we use the base class property here?

Reply:
Hmm well python handles the message a little bit different tho take a look here: https://stackoverflow.com/questions/4690600/python-exception-message-capturing

🤔

Reply:
@OGKevin thanks for explanation! Let's keep it as we have it now :)